### PR TITLE
Adds "gluster volume status" to CLI

### PR DESCRIPTION
**Pull request by nnDarshan**
This Patch adds gluster volume status to cli. It
takes volume-name as optional argument. If volume
name not provided the volume status for all the
volumes will be displayed.

Issue: https://github.com/gluster/glusterd2/issues/488
Signed-off-by: Darshan N <darshan.n.2024@gmail.com>

### CODE REVIEW

**Comment by nnDarshan** [comment]
@prashanthpai @aravindavk @kshlm @atinmu Please review

**Comment by nnDarshan** [comment]
> Can you please add a e2e test for this change as well?

@atinmu This already seems to be there: https://github.com/gluster/glusterd2/blob/master/e2e/volume_ops_test.go#L143
Is this waht you are expecting ?

**Comment by atinmu** [comment]
> @atinmu This already seems to be there: https://github.com/gluster/glusterd2/blob/master/e2e/volume_ops_test.go#L143 . Is this waht you are expecting ?

@nnDarshan is it possible to validate the CLI functionality itself in a e2e test?

**Comment by prashanthpai** [comment]
> @nnDarshan is it possible to validate the CLI functionality itself in a e2e test?

The scope of e2e framework is limited to functional testing of REST API and the `restclient` package. The CLI binary is a [very thin wrapper](https://github.com/gluster/glusterd2/blob/master/glustercli/cmd/volume.go#L312) around `restclient` package. The e2e tests are already covering testing of `restclient` package.

Further, CLI doesn't support json output (no need for it yet as REST supports it). Any tests for CLI at this point in time has to parse raw output - some outputs are plain text and some are in table form and writing that logic is counter-productive. We can focus on testing `restclient` package instead.

**Comment by atinmu** [comment]
> The scope of e2e framework is limited to functional testing of REST API and the restclient package. The CLI binary is a very thin wrapper around restclient package. The e2e tests are already covering testing of restclient package. Further, CLI doesn't support json output (no need for it yet as REST supports it). Any tests for CLI at this point in time has to parse raw output - some outputs are plain text and some are in table form and writing that logic is counter-productive. We can focus on testing restclient package instead.

Fair point.

**Comment by atinmu** [comment]
@nnDarshan can you please rebase the PR?

**Comment by nnDarshan** [comment]
> @nnDarshan can you please rebase the PR?

@atinmu Done

**Comment by atinmu** [comment]
Looks good to me. Passing it to @prashanthpai @aravindavk

File: glustercli/cmd/volume.go [hunk 1]
```diff
@@ -352,16 +353,62 @@ var volumeListCmd = &cobra.Command{
...
 	Use:   "status",
 	Short: helpVolumeStatusCmd,
 	Args:  cobra.RangeArgs(0, 1),
 	Run: func(cmd *cobra.Command, args []string) {
-		volname := "all"
-		if len(cmd.Flags().Args()) > 0 {
-			volname = cmd.Flags().Args()[0]
+		err := volumeStatusHandler(cmd)
+		if err != nil {
+			log.Println("No volumes present")
```
Comment (prashanthpai) [inline glustercli/cmd/volume.go]: nit: no volumes present is misleading. You can remove it or change it to "volume status failed"

Comment (nnDarshan) [reply glustercli/cmd/volume.go]: Done

**Pull request merged**